src/WebJobs.Script/Utility.cs

```diff
@@ -81,17 +81,55 @@ internal static async Task InvokeWithRetriesAsync(Func<Task> action, int maxRetr
         /// <param name="pollingIntervalMilliseconds">The polling interval.</param>
         /// <param name="condition">The condition to check</param>
         /// <returns>A Task representing the delay.</returns>
-        internal static async Task DelayAsync(int timeoutSeconds, int pollingIntervalMilliseconds, Func<bool> condition)
+        internal static Task<bool> DelayAsync(int timeoutSeconds, int pollingIntervalMilliseconds, Func<bool> condition)
         {
+            return DelayAsyncHelper(timeoutSeconds, pollingIntervalMilliseconds, condition, null, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Delays while the specified condition remains true.
+        /// </summary>
+        /// <param name="timeoutSeconds">The maximum number of seconds to delay.</param>
+        /// <param name="pollingIntervalMilliseconds">The polling interval.</param>
+        /// <param name="condition">The async condition to check</param>
+        /// <returns>A Task representing the delay.</returns>
+        internal static Task<bool> DelayAsync(int timeoutSeconds, int pollingIntervalMilliseconds, Func<Task<bool>> condition, CancellationToken cancellationToken)
+        {
+            return DelayAsyncHelper(timeoutSeconds, pollingIntervalMilliseconds, null, condition, cancellationToken);
+        }
+
+        private static async Task<bool> DelayAsyncHelper(int timeoutSeconds, int pollingIntervalMilliseconds, Func<bool> condition, Func<Task<bool>> asyncCondition, CancellationToken cancellationToken)
```

Comment:
We can have a single condition parameter here and always expect it to be a task returning one. We just need to make sure the overload that takes a `Func<bool>` correctly wraps the delegate and calls this overload with that.

Reply:
thanks. Fixed it.